rfcs/features/0041-renderer-plugin/README.md

```diff
@@ -0,0 +1,209 @@
+# 0041: Renderer Plugin
+- Authors: [Richard Spellman](namlleps.drahcir@gmail.com)
+- Status: [PROPOSED](/README.md#proposed)
+- Since: 2023-05-24
+- Status Note: This is just a stub, requires initial content.
+- Supersedes: (link to anything this RFC supersedes)
+- Start Date: 2023-05-24
+- Tags: [feature](/tags.md#feature)
+
+## Summary
+
+The approach to renderering credentials in a human friendly way is still evolving, but there is an emerging approach that is becoming stable enough to begin to implement. This component is a vc-kit plugin that takes a verifiable credential (perhaps, later, also a verifiable presentation) and returns a human friendly rendering of the credential. 
+
+## Motivation
+
+vc-kit is positioned to provide a verifiable credentials solution in the context of regulated industry. As such, there is commonly a requirement, on the part of issuers, that their credentials are presented in a pre-defined format. This puts a requirement on wallets to render credentials so that they comply with this format. 
+
+In the context of transitioning to a fully decentralised, digital ecosystem, there remains the requirement for credentials to be transmitted in their rendered format - either via email, or printed documents. 
+
+## Tutorial
+
+The Verifiable Credential Data Model v2.0 indroduces a new, reserved term 'renderMethod' which can be used by wallets to display human friendly renderings of credentials.
+
+Let's assume we have this credential, presented here without proof for simplicity.
+
+```json
+{
+   "@context": [
+    "https://www.w3.org/2018/credentials/v1",
+    "https://www.w3.org/2018/credentials/examples/v1",
+     {
+       "render":"https://www.w3.org/2018/credentials#renderMethod"
+      }
+  ],
+  "id": "http://example.edu/credentials/3732",
+  "type": ["VerifiableCredential", "UniversityDegreeCredential"],
+  "issuer": "https://example.edu/issuers/565049",
+  "issuanceDate": "2010-01-01T00:00:00Z",
+  "credentialSubject": {
+    "id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
+    "name": "Jane Smith",
+    "degree": {
+      "type": "BachelorDegree",
+      "name": "Bachelor of Science and Arts",
+      "institution": "Example University"
+    }
+  },
+  "render": [{
+    "@id": "<div style=\"width:300px; height:100px; border: 2px solid black; text-align:center\">\n  <div>\n    This {{credentialSubject.degree.name}} is conferred to\n  </div>\n  <strong style=\"font-size: 16px\">\n    {{credentialSubject.name}}\n  </strong>\n  <div>\n    by {{credentialSubject.degree.institution}}.\n  </div>\n</div>",
```

Comment:
`{{credentialSubject.degree.name}}` the syntax here is using the handlebars syntax, is it right?
Do we support another templating language?

Reply:
eventually, the syntax will be defined by the spec for the specific renderMethod object. at the moment it is enough to assume handlebars in each of the example methods I gave in the feature (e.g. svg, html). these are draft, example render methods and don't have a formal specification. we are setting up a structure to adapt to render method specs as they emerge - hence the function with single responsibility for a given render method, and ability to pass in via agent configuration.